src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -103,21 +104,28 @@ public ImmutableArray<PackageSource> GetPackageSources()
                 return packageSources;
             }
 
-            try
+            lock (_gate)
             {
-                packageSources = _packageSourceProvider.Value.GetSources(includeUnOfficial: true, includeDisabled: false)
-                    .SelectAsArray(r => new PackageSource(r.Key, r.Value));
+                if (_packageSourcesAsync is null)
```

Comment:
`#nullable enable` the file?

Reply:
Leaving this one for a future change. The one case where it looked like it could have helped us isn't caught by the compiler anyway.